### PR TITLE
Add GetResourceResultsByRuleCategory

### PR DESCRIPTION
**Pull request by GMadorell**
[sc-7334]

### CODE REVIEW

File: com/coralogix/xdr/v1/test/security_test_service.proto [hunk 1]
```diff
@@ -78,6 +78,14 @@ service SecurityTestService {
       body: "*"
     };
   }
+  
+  rpc GetAssetResultsByRuleCategory(GetAssetResultsByRuleCategoryRequest) returns (GetAssetResultsByRuleCategoryResponse) {
+    option (audit_log_description).description = "get asset test results grouped by rule category";
+    option (google.api.http) = {
+      post: "/xdr/v1/security-test/group-by-rule-category-and-asset"
```
Comment (xBATx) [inline com/coralogix/xdr/v1/test/security_test_service.proto:85]: Just wondering if we shouldn't create a new `asset_service.proto` for this as we're going to have more functionality so it'll be good to have it grouped in the new service.

File: com/coralogix/xdr/v1/test/security_test_service.proto [hunk 2]
```diff
@@ -175,3 +183,24 @@ message GetSecurityTestResultsByPolicyCategoryResponse {
...
 }
+
+message GetAssetResultsByRuleCategoryRequest {
+  optional google.protobuf.StringValue execution_id = 1;
+  optional com.coralogix.xdr.v1.filter.SecurityTestFilter filter = 2;
+}
+
+message GetAssetResultsByRuleCategoryResponse {
+  message AssetResults {
+    google.protobuf.StringValue asset_title = 1;
```
Comment (xBATx) [inline com/coralogix/xdr/v1/test/security_test_service.proto:194]: Hmm I'd rename `asset_title` to `asset_name` as:
* we don't have a title and name for an asset in the logs - only the asset name
* the name will be used to navigate also to the detail and title sounds just confusing.

File: com/coralogix/xdr/v1/test/security_test_service.proto (hunk 1 above)
Comment (GMadorell) [reply com/coralogix/xdr/v1/test/security_test_service.proto:85]: Hmm, I think it makes sense to put it in the same service, as we are just returning test results, right? It's just a different way to organize them.
Alternatively, we can create a whole new service only for this but we would have to abstract quite a bit of pieces of logic that we have already implemented in the existing service.
I'd say we can put it here for now and we can always abstract it later on if we have different functionalities related to assets.

File: com/coralogix/xdr/v1/test/security_test_service.proto (hunk 2 above)
Comment (GMadorell) [reply com/coralogix/xdr/v1/test/security_test_service.proto:194]: Yeah good point, I think it makes sense renaming to "name". It's going to be a textual identifier for now but we can always refactor later 👍 .

**Review by xBATx — COMMENTED** [review]
LGTM in general but let's wait till tomorrow as it's possible that we would need a new separate tab for asset explorer. 
If it's so, I'd vote for creating a new service for this functionality. Let's sync tomorrow on standup.

**Review by xBATx — APPROVED** [review]
Awesome 🎉

**Pull request merged**